x-pack/plugin/async-search/src/internalClusterTest/java/org/elasticsearch/xpack/search/CrossClusterAsyncSearchIT.java

```diff
@@ -1849,8 +2061,16 @@ public void onNewReaderContext(ReaderContext readerContext) {
 
                 @Override
                 public void onPreQueryPhase(SearchContext searchContext) {
-                    startedLatch.get().countDown();
-                    final CountDownLatch latch = queryLatch.get();
+                    final CountDownLatch latch;
+                    if (searchContext.indexShard().shardId().getIndexName().equals("remote")) {
+                        startedRemoteLatch.get().countDown();
+                        latch = remoteQueryLatch.get();
+                    } else if (searchContext.indexShard().shardId().getIndexName().equals("local")) {
+                        startedLocalLatch.get().countDown();
+                        latch = localQueryLatch.get();
```

Comment:
By introducing a distinction between local and remote here, I can block local or remote query phase selectively.